src/api/Folder.js

```diff
@@ -90,6 +104,64 @@ class Folder extends Item {
         return cache.has(this.key);
     }
 
+    /**
+     * Takes representations fields from items in collection and assigns thumbnailUrls
+     * to items, with the use of the callback
+     *
+     * @param {Collection} collection - collection with items that have representations field populated
+     * @param {Token} token - access_token
+     * @param {(item: BoxItem, index: number, thumbnailUrl: string) => void} callback - Function to call with generated thumbnailUrl
+     * @return {void}
+     */
+    formatRepresentations(
+        collection: Collection,
+        token: Token,
+        callback: (item: BoxItem, index: number, thumbnailUrl: string) => void,
+    ): void {
+        if (collection.items) {
+            collection.items.forEach(async (item, index) => {
+                const entries = getProp(item, 'representations.entries');
+                if (getProp(entries, '[0].status.state') === 'success') {
+                    const { representation, template } = this.getDataFromEntries(entries, 0);
+                    if (!!template && !!representation) {
+                        const tokenString = await TokenService.getToken(getTypedFileId(item.id), token);
```

Comment:
I think it's possible for the token to come back as an object as well, see https://github.com/box/box-ui-elements/blob/e282d5f53e3d0e3c15a728e9b113888269acfea6/src/utils/TokenService.js#L64